pkg/apis/scorecard/v1alpha2/types.go

```diff
@@ -34,16 +34,18 @@ const (
 
 // ScorecardTestResult contains the results of an individual scorecard test.
 type ScorecardTestResult struct {
-	// Name is the name of the test
+	// Name is the name of the test image
 	Name string `json:"name"`
 	// Description describes what the test does
-	Description string `json:"description"`
+	Description string `json:"description,omitempty"`
```

Comment:
Above we have `Description` replaced by `EntryPoint`, so should not this change be reflected here as well?

Reply:
Again, This struct might be used somewhere as well, Hence only adding the field `EntryPoint` . The requirement as I understand is only to remove from scorecard output format for `--list` sub command.

Reply:
This raises an issue that @jmccormick2001 and I missed.

Right now, the function signatures for List and Run both return a slice of test results, so we clearly need to change something about those things to cover the fact that listing the tests that _will_ run returns less information than actually running the tests.

We probably need to jump on a call to nail this down when @jmccormick2001 gets back, but my initial idea is to do something like this:

```go
type TestSpec struct {
  Image      string
  Entrypoint []string
  Labels     map[string]string
}

type ScorecardTestResult struct {
  Name        string
  Description string
  State       State
  Errors      []string
  Suggestions []string
}
```

And then decide how to return these from `List` and `Run`. List could return `TestSpec` and Run could return `ScorecardTestResult`, or we could have a `Test` type that embeds these types, and then `List` and `Run` could both return `Test` where `List` only populates the `TestSpec` portion.

Also related to this is the idea of test images returning multiple results.

To unblock this for now, I would suggest that we have `List()` return `[]TestSpec` and `Run()` return `[]ScorecardTestResult`

Reply:
Okay, Also `Labels` field is being used along with rest of the fields in `ScorecardTestResult`, Example here: https://github.com/operator-framework/operator-sdk/blob/master/pkg/apis/scorecard/v1alpha2/formatter.go#L47 
Do we want to change how this is implemented as well with `Labels` being moved to `TestSpec` struct?

Reply:
Oh right. So maybe we do need to go ahead and introduce a parent `Test` struct.

```go
type Test struct {
   TestSpec
   Results []ScorecardTestResult
}
```

Also something very similar to the `TestSpec` struct likely already exists to load the config file. We should take a holistic look at this and not duplicate existing structures.

Reply:
https://github.com/operator-framework/operator-sdk/blob/b7c88773342d79a5500120fed5efb9b86936ea5b/internal/scorecard/alpha/config.go#L23